### PR TITLE
Refactoring tutorials for the new changes in the prometheus-service

### PR DESCRIPTION
**Pull request by rajdas98**
Refactoring doc for the new changes in the prometheus-service
Issue- https://github.com/keptn-contrib/prometheus-service/issues/53

Signed-off-by: Raj Das <mail.rajdas@gmail.com>

### CODE REVIEW

**Comment by jetzlstorfer** [comment]
It looks like something is broken. I looked at the preview and the tutorial has some steps missing. The root cause here is most probably that the formatting is broken and the next pages could not be rendered.
image

**Comment by jetzlstorfer** [comment]
Once this tutorial is fixed, we also need to replicate the updates in the other tutorial, such as
- https://tutorials.keptn.sh/tutorials/keptn-multistage-qualitygates-08/index.html?index=..%2F..index#0 
- https://tutorials.keptn.sh/tutorials/keptn-litmus-08/index.html?index=..%2F..index#0

File: site/tutorials/snippets/08/monitoring/setupPrometheus.md [hunk 1]
```diff
@@ -2,39 +2,100 @@
 ## Setup Prometheus Monitoring
 Duration: 3:00
 
-After creating a project and service, you can setup Prometheus monitoring and configure scrape jobs using the Keptn CLI. 
+After creating a project and service, you can set up Prometheus monitoring and configure scrape jobs using the Keptn CLI.
 
-To install the *prometheus-service*, execute: 
+Keptn doesn't install or manage Prometheus and its components. Users need to install Prometheus and Prometheus Alert manager as a prerequisite.
+
```
Comment (jetzlstorfer) [inline site/tutorials/snippets/08/monitoring/setupPrometheus.md:8]: Please add a sentence here for the Prometheus installation with helm and the community-managed helm chart that you are using. The purpose of the tutorial is that the user can follow all steps and at the end has a working environment. 
Therefore, also the steps to install Prometheus are needed.

**Review by jetzlstorfer — CHANGES_REQUESTED** [review]

File: site/tutorials/snippets/08/monitoring/setupPrometheus.md (hunk 1 above)
Comment (rajdas98) [reply site/tutorials/snippets/08/monitoring/setupPrometheus.md:8]: Done

File: site/tutorials/keptn-litmus-08.md [hunk 2]
```diff
@@ -86,23 +86,88 @@ Duration: 3:00
...
+* To install the prometheus-service, execute:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/keptn-contrib/prometheus-service/release-0.5.0/deploy/service.yaml
+```
+
+* Edit the prometheus-service deployment and patch it with these environment variables:
+
+```bash
+kubectl edit deploy prometheus-service -n keptn
```
Comment (jetzlstorfer) [inline site/tutorials/keptn-litmus-08.md:116]: we can make this easier for the user with the following commands:
```
kubectl set env deployment/prometheus-service -n keptn --containers="prometheus-service" PROMETHEUS_NS=monitoring
kubectl set env deployment/prometheus-service -n keptn --containers="prometheus-service" ALERT_MANAGER_NS=monitoring
```

Comment (jetzlstorfer) [reply site/tutorials/keptn-litmus-08.md:116]: with this we can ensure that users do not make a typo as they can copy and paste both commands.

Comment (jetzlstorfer) [reply site/tutorials/keptn-litmus-08.md:116]: Please update the text, with something like:

Next, we are going to set the namespace that Prometheus has been deployed (i.e., the `monitoring` namespace) in the environment variables section of Keptn's `prometheus-service`.

File: site/tutorials/keptn-multistage-qualitygates-08.md [hunk 3]
```diff
@@ -213,46 +213,87 @@ Navigating to the URLs should result in the following output:
...
-bash wait_for_deployment_in_namespace "prometheus-service-monitoring-configure-distributor" "keptn" 
-sleep 10
--->
-    
-Negative
-: Please note that a Keptn managed Prometheus will be installed only after executing the next command. At this step, we have only installed the service that is responsible for managing Prometheus.
+* Edit the prometheus-service deployment and patch it with these environment variables:
 
-Execute the following command to install Prometheus and set up the rules for the *Prometheus Alerting Manager*:
+```bash
```
Comment (jetzlstorfer) [inline site/tutorials/keptn-multistage-qualitygates-08.md:242]: Please align the instructions to the previous comment

File: site/tutorials/snippets/08/monitoring/setupPrometheus.md [hunk 4]
```diff
@@ -1,40 +1,109 @@
...
+bash wait_for_deployment_in_namespace "prometheus-service" "keptn" 
+bash wait_for_deployment_in_namespace "prometheus-service-monitoring-configure-distributor" "keptn" 
+sleep 10
+-->
+
+* Edit the prometheus-service deployment and patch it with these environment variables:
+
+<!-- command -->
+```
+kubectl edit deploy prometheus-service -n keptn
```
Comment (jetzlstorfer) [inline site/tutorials/snippets/08/monitoring/setupPrometheus.md:36]: Align to the previous comment

File: site/tutorials/snippets/08/monitoring/setupPrometheus.md [hunk 5]
```diff
@@ -1,40 +1,109 @@
...
+sleep 10
+-->
 
-### Optional: Verify Prometheus setup in your cluster
+#### Optional: Verify Prometheus setup in your cluster
 * To verify that the Prometheus scrape jobs are correctly set up, you can access Prometheus by enabling port-forwarding for the prometheus-service:
 
     ```
-    kubectl port-forward svc/prometheus-service 8080 -n monitoring
+    kubectl port-forward svc/prometheus-service 8080 -n <PROMETHEUS_NS>
```
Comment (jetzlstorfer) [inline site/tutorials/snippets/08/monitoring/setupPrometheus.md:106]: should be `kubectl port-forward svc/prometheus-server 8080:80 -n monitoring`

File: site/tutorials/keptn-litmus-08.md [hunk 6]
```diff
@@ -86,23 +86,88 @@ Duration: 3:00
...
+
+```bash
+keptn configure monitoring prometheus --project=sockshop --service=carts
+```
+
+#### Optional: Verify Prometheus setup in your cluster
+* To verify that the Prometheus scrape jobs are correctly set up, you can access Prometheus by enabling port-forwarding for the prometheus-service:
+
+```BASH
+kubectl port-forward svc/prometheus-service 8080 -n monitoring
```
Comment (jetzlstorfer) [inline site/tutorials/keptn-litmus-08.md:166]: kubectl port-forward svc/prometheus-server 8080:80 -n monitoring

File: site/tutorials/keptn-multistage-qualitygates-08.md [hunk 7]
```diff
@@ -213,46 +213,87 @@ Navigating to the URLs should result in the following output:
...
 
-- To verify that the Prometheus scrape jobs are correctly set up, you can access Prometheus by enabling port-forwarding for the prometheus-service:
+#### Optional: Verify Prometheus setup in your cluster
+* To verify that the Prometheus scrape jobs are correctly set up, you can access Prometheus by enabling port-forwarding for the prometheus-service:
 
-    ```
-    kubectl port-forward svc/prometheus-service 8080 -n monitoring
-    ```
+```BASH
+kubectl port-forward svc/prometheus-service 8080 -n monitoring
```
Comment (jetzlstorfer) [inline site/tutorials/keptn-multistage-qualitygates-08.md:293]: please double check and see other comments on this

**Review by jetzlstorfer — CHANGES_REQUESTED** [review]

**Comment by jetzlstorfer** [comment]
Please remove the instructions that are not necessary from this step and only keep the ones that need to be executed.

image

**Comment by jetzlstorfer** [comment]
We'll wait for the merge of Prometheus-service and Prometheus-sli-service to be completed and then will upgrade this tutorial to reflect the new changes

File: site/tutorials/keptn-multistage-qualitygates-08.md [hunk 8]
```diff
@@ -231,41 +261,30 @@ bash wait_for_deployment_in_namespace "prometheus-service-monitoring-configure-d
...
-Negative
-: Please note that a Keptn managed Prometheus will be installed only after executing the next command. At this step, we have only installed the service that is responsible for managing Prometheus.
-
-Execute the following command to install Prometheus and set up the rules for the *Prometheus Alerting Manager*:
 
+* Execute the following command to install Prometheus and set up the rules for the *Prometheus Alerting Manager*:
 <!-- command -->
 ```
-keptn configure monitoring prometheus --project=pod-tato-head --service=helloservice
+keptn configure monitoring prometheus --project=sockshop --service=carts
```
Comment (jetzlstorfer) [inline site/tutorials/keptn-multistage-qualitygates-08.md:268]: please change back to original project name

**Review by jetzlstorfer — APPROVED** [review]
lgtm

**Pull request merged**